js/src/main/scala/java/util/TimeZone.scala

```diff
@@ -84,7 +92,22 @@ abstract class TimeZone extends Serializable with Cloneable {
   def getDisplayName(daylight: Boolean, style: Int, locale: Locale): String = {
     if (style != TimeZone.SHORT || style != TimeZone.LONG)
       throw new IllegalArgumentException(s"Illegal timezone style: $style")
-    ???
+
+    val id = getID
+    val zoneName = DateFormatSymbols.getInstance(locale).getZoneStrings.find(_(0) == id).flatMap { strs =>
```

Comment:
The call `_(0)` could raise an exception. Could you handle it, e.g. filtering out empty arrays?